### PR TITLE
:wrench: Maintenance and small improvements

### LINKED ISSUE #113: :bug:Training with WideResNetBaseline
hey!

First of all, I really want to appreciate for creating this amazing library!

When I try to apply the WideResNetBaseline for image classification task, the testing accuracy is really low. This circumstance appears in different datasets including MNIST and CIFAR. The codes and results are provided below:

```
from pathlib import Path
from torch_uncertainty.utils import TUTrainer
from torch_uncertainty.datamodules import MNISTDataModule, CIFAR10DataModule
from torch_uncertainty.baselines.classification import WideResNetBaseline
import torch

trainer = TUTrainer(accelerator="cuda", max_epochs=10, enable_progress_bar=True)

root = Path("/home/torch_uncertainty/data")
datamodule = MNISTDataModule(root=root, batch_size=128, eval_ood=False)
loss_fn = torch.nn.CrossEntropyLoss()
model = WideResNetBaseline(datamodule.num_classes,
                           datamodule.num_channels,
                           loss_fn,
                           style="cifar",
                           version="std")

trainer.fit(model=model, datamodule=datamodule)
trainer.test(model=model, datamodule=datamodule)
```
```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│       test/cal/ECE        │          0.02958          │
│       test/cal/aECE       │          0.03167          │
│       test/cls/Acc        │          0.10580          │
│      test/cls/Brier       │          0.90634          │
│       test/cls/NLL        │          2.33843          │
│     test/cls/entropy      │          2.27538          │
│       test/sc/AUGRC       │          0.44700          │
│       test/sc/AURC        │          0.89636          │
│    test/sc/CovAt5Risk     │            nan            │
│    test/sc/RiskAt80Cov    │          0.89113          │
└───────────────────────────┴───────────────────────────┘
```
Since the Tutorials you provided in the documents does not cover the usage of "torch_uncertainty.baselines" yet, I wonder if I have made some mistakes in my codes?

Besides, I think it will be better to monitor the training process if there exists parameter in trainer.fit() for metrics presentation.

Looking forward to your guidance and thanks again!

### PR DESCRIPTION
**Pull request by o-laurent**
Small maintenance pull request. 
- Support the last version of the HF CLI
- Improve the logging of important metrics in the progress bar
- Add `optim_recipe` arg to the baselines when not using the CLI (thanks to #113)
- Fix quickstart 

@alafage, should we make this 0.2.2.post0 to propagate the fixes? Notably, 0.2.2 probably does not work anymore due to HF's update. If so, we need to :zap: the package version.

Fixes #113

### CODE REVIEW

**Comment by alafage** [comment]
> @alafage, should we make this 0.2.2.post0 to propagate the fixes? Notably, 0.2.2 probably does not work anymore due to HF's update. If so, we need to ⚡ the package version.

Yes, that's best!

**Review by alafage — APPROVED** [review]
